File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
using namespace std;
int dat[10000001]={0};
int cnt=0;
int solution(int k, vector<int> tangerine) {
    for(int i=0;i<tangerine.size();i++){
        dat[tangerine[i]]++;
    }
    sort(dat,dat+10000001, greater<int>());
    
    int sum=0;
    for(int i=0;i<10000001;i++){
        if(sum>k){
            return cnt;
        }
        else if(sum==k){
            return cnt;
        }
        else{
            cnt++;
            sum+=dat[i];
        }
    }
    
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;
class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        Map<Integer, Integer> map = new HashMap<>();
        for(int i=0;i<tangerine.length;i++){
            if(map.get(tangerine[i]) == null) map.put(tangerine[i], 1);
            else{
                int val = map.get(tangerine[i]);
                map.put(tangerine[i], val + 1);
            }
        }
        List<Integer> list = new ArrayList<>();
        for(Integer key: map.keySet()){
            list.add(map.get(key));
        }
        Collections.sort(list, Collections.reverseOrder());
        int tmp = 0;
        for(int i=0;i<list.size();i++){
            if(tmp < k){
                answer++;
                tmp += list.get(i);
            }
        }
        return answer;
    }
}